controllers/stargate/stargate_telemetry_reconciler_test.go

```diff
@@ -0,0 +1,55 @@
+// Some of the tests in this package target functions from pkg/telemetryapi, because they need to be envtests. We prefer to keep envtests in the controller
+// packages.
+package stargate
+
+import (
+	"context"
+	"testing"
+
+	testlogr "github.com/go-logr/logr/testing"
+	telemetryapi "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/config"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/test"
+	promapi "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// new_DummyK8ssandraClusterReconciler gives us back a `K8ssandraClusterReconciler` with just the fields that we need to test `reconcileCassandraDCTelemetry()`.
+func newDummyK8ssandraClusterReconciler() StargateReconciler {
+	return StargateReconciler{ReconcilerConfig: &config.ReconcilerConfig{DefaultDelay: interval}}
+}
+
+// Test_reconcileStargateTelemetry_TracksNamespaces tests that the servicemonitor is created
+func Test_reconcileStargateTelemetry_succeeds(t *testing.T) {
```

Comment:
We are relying on `envtest` for controller integration tests. We need a new or updated integration test for the Stargate controller (see `stargate_controller_test.go`) to verify the ServiceMonitor functionality. 

It would be good to have a happy path test which will of course require having the ServiceMonitor CRD installed. We configure CRDs in `testenv.go`. Assuming you already have unit test coverage for creating the ServiceMonitor, it would be probably be sufficient to check that the object was actually created.

It would be nice to also verify that the ServiceMonitor is deleted when telemetry is disabled.

Reply:
Also done.